src/SparkTestBase.sol

```diff
@@ -158,29 +301,47 @@ abstract contract SparklendTestBase is ProtocolV3TestBase, SparkTestBase {
         }
     }
 
-    function testE2E() public {
+    function test_ETHEREUM_E2E() public {
+        testE2E(ChainIdUtils.Ethereum());
+    }
+
+    function test_GNOSIS_E2E() public {
+        vm.skip(chainSpellMetadata[ChainIdUtils.Gnosis()].payload == address(0));
+        testE2E(ChainIdUtils.Gnosis());
+    }
+
+    function testE2E(ChainId chainId) internal onChain(chainId) {
         if (disableE2E) return;
 
-        address[] memory poolProviders = poolAddressesProviderRegistry.getAddressesProvidersList();
+        address[] memory poolProviders = _getPoolAddressesProviderRegistry().getAddressesProvidersList();
 
         for (uint256 i = 0; i < poolProviders.length; i++) {
             loadPoolContext(poolProviders[i]);
             e2eTest(pool);
         }
 
-        executePayload(payload);
+        executeAllPayloadsAndBridges();
```

Comment:
I would make an exception for Ethereum and only execute the mainnet payload without triggering any cross-chain messages to fix the OOG error for now.